app/compare/compare_invocations.tsx

```diff
@@ -87,11 +122,30 @@ export default class CompareInvocationsComponent extends React.Component<Compare
     this.setState({ showChangesOnly: !this.state.showChangesOnly });
   }
 
+  private onClickPreProcessingOption(e: React.ChangeEvent<HTMLInputElement>) {
+    const { name, checked } = e.target;
+    const preProcessingOptions = { ...this.preProcessingOptions, [name]: checked };
+    router.replaceParams({ options: JSON.stringify(preProcessingOptions) });
```

Comment:
putting json into the url feels kinda gross, any way we could easily map these to query params?